auslib/db.py

```diff
@@ -928,16 +929,20 @@ def __init__(self, db, dialect, metadata, baseTable, conditions=("time", "uptake
             # See https://github.com/zzzeek/sqlalchemy/blob/rel_0_7/lib/sqlalchemy/schema.py#L781
             # for background.
             newcol.key = newcol.name = "base_%s" % col.name
-            # 2) Primary keys on the base table are normal columns here,
-            # and are allowed to be null (because we support adding new rows
-            # to tables with autoincrementing keys via scheduled changes).
-            # The base table's data version may also be null for the same reason.
-            if col.primary_key or newcol.name == "base_data_version":
+            # 2) All non PK columns from baseTable should be nullable. Base table PK columns
+            # which are autoincrement should be nullable in the Scheduled Changes version of
+            # the column. Base table PK columns which are not autoincrement should not be nullable.
```

Comment:
Yes, it's a huge help! Let me run an updated version through you to see if I got this correct:

Primary Key Integer Autoincrement columns from the baseTable become normal nullable
columns because we can schedule changes that insert into baseTable and the DB will handle
inserting the correct value. However, nulls aren't allowed when we schedule updates or deletes - this is enforced in self.validate().
For Primary Key columns that aren't Integer or Autoincrement but are nullable, we preserve
this non-nullability because we need a value to insert into the baseTable when the
scheduled change gets executed.
Non-Primary Key columns from the baseTable become nullable and non-unique because a
scheduled deletion will have null values in all non-PK even if they are non-nullable or
unique in the baseTable.